dht_bootstrap.go

```diff
@@ -91,11 +125,35 @@ func (dht *IpfsDHT) startRefreshing() error {
 	return nil
 }
 
+func collectWaitingChannels(source chan chan<- error) []chan<- error {
+	var waiting []chan<- error
+	for {
+		select {
+		case res := <-source:
+			if res != nil {
+				waiting = append(waiting, res)
+			}
+		default:
+			return waiting
+		}
+	}
+}
+
 func (dht *IpfsDHT) doRefresh(ctx context.Context) error {
 	var merr error
-	if err := dht.selfWalk(ctx); err != nil {
-		merr = multierror.Append(merr, err)
+
+	// wait for self walk result
+	selfWalkres := make(chan error, 1)
+	dht.triggerSelfLookup <- selfWalkres
```

Comment:
shouldn't this be in a select statement with ctx.Done?

Reply:
Done.